### ISSUE TITLE
Type signature of combine

### ISSUE THREAD

**Issue by AlexGalays** [staltz/xstream#28]
It seems combine has a wrong type signature, at least when used statically from the Stream object.

Using it with just two streams, the project function expects 3 arguments?

```
import xs from 'xstream'

const stream = Stream.create<string>({
	start: listener => {},
	stop: () => {}
})

const stream2 = Stream.create<string>({
	start: listener => {},
	stop: () => {}
})

const combined = Stream.combine(
	(a, b) => a.slice(2) + b.slice(2),
	stream, stream2
)
```

This doesn't compile, because the `a` passed to the project function is inferred to be an object.

**Comment by staltz** [comment]
For the static combine, use `xs.combine`, not `Stream.combine`.

`xs.combine(fn, a, b)` then fn expects 2 args

`streamInstance.combine(fn, a, b)` then fn expects 3 args.

Does this solve your case?

**Comment by AlexGalays** [comment]
Isn't `xs` and `Stream` exactly the same?

In my own code, I use 

```
import xs from 'xstream'
xs.combine...
```

Only had to use `Stream` on the typescript playground.

**Comment by staltz** [comment]
Ok, sorry, I think it's actually incorrect in code. I'll fix this.

**Comment by staltz** [comment]
Released v2.1.2 with a fix
